aviatrix/data_source_aviatrix_vpc.go

```diff
@@ -197,9 +202,32 @@ func dataSourceAviatrixVpcRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("aviatrix_firenet_vpc", false)
 	}
 
-	d.Set("vpc_id", vC.VpcID)
-	if vC.CloudType == goaviatrix.AZURE {
-		d.Set("resource_group", strings.Split(vC.VpcID, ":")[1])
+	if vC.CloudType == goaviatrix.GCP {
+		d.Set("vpc_id", strings.Split(vC.VpcID, "~-~")[0])
+	} else {
+		d.Set("vpc_id", vC.VpcID)
+		if vC.CloudType == goaviatrix.AZURE {
+			account := &goaviatrix.Account{
+				AccountName: d.Get("account_name").(string),
+			}
+
+			acc, err := client.GetAccount(account)
+			if err != nil {
+				if err != goaviatrix.ErrNotFound {
+					return fmt.Errorf("aviatrix Account: %s", err)
+				}
+			}
+
+			var subscriptionId string
+			if acc != nil {
+				subscriptionId = acc.ArmSubscriptionId
+			}
+
+			resourceGroup := strings.Split(vC.VpcID, ":")[1]
+			azureVnetResourceId := "/subscriptions/" + subscriptionId + "/resourceGroups/" + resourceGroup + "/providers/Microsoft.Network/virtualNetworks/" + vC.Name
+			d.Set("resource_group", resourceGroup)
+			d.Set("azure_vnet_resource_id", azureVnetResourceId)
+		}
```

Comment:
This block does not need to be in the `else`. Less indentation the better.

Reply:
also refactored this part in resource_aviatrix_vpc